openmc/particle_restart.py

```diff
@@ -0,0 +1,92 @@
+import h5py
+
+import openmc.checkvalue as cv
+
+_VERSION_PARTICLE_RESTART = 2
+
+class Particle(object):
+    """Information used to restart a specific particle that caused a simulation to
+    fail.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the particle restart file
+
+    Attributes
+    ----------
+    current_batch : int
+        The batch containing the particle
+    generations_per_batch : int
+        Number of generations per batch
+    current_generation : int
+        The generation containing the particle
+    n_particles : int
+        Number of particles per generation
+    run_mode : int
+        Type of simulation (criticality or fixed source)
+    id : long
+        Identifier of the particle
+    type : int
+        Particle type (1 = neutron, 2 = photon, 3 = electron, 4 = positron)
+    weight : float
+        Weight of the particle
+    energy : float
+        Energy of the particle in eV
+    xyz : list of float
+        Position of the particle
+    uvw : list of float
+        Directional cosines of the particle
+
+    """
+
+    def __init__(self, filename):
+        self._f = h5py.File(filename, 'r')
+
+        # Ensure filetype and version are correct
+        cv.check_filetype_version(self._f, 'particle restart',
+                                  _VERSION_PARTICLE_RESTART)
+
+    @property
+    def current_batch(self):
```

Comment:
Is it OK if any of these don't exist in `self._f`? You might want to use `.get()` instead to supply a default case.